node/src/main/scala/coop/rchain/node/api/DeployGrpcService.scala

```diff
@@ -24,7 +25,10 @@ private[api] object DeployGrpcService {
     new CasperMessageGrpcMonix.DeployService {
 
       private def defer[A](task: F[A]): Task[A] =
-        Task.defer(task.toTask).executeOn(worker)
+        Task.defer(task.toTask).executeOn(worker).attempt.flatMap {
```

Comment:
Is that a case for `adaptError`?

Reply:
I'll look into this later. Will merge for now.